### PR TITLE
Adding missing readme files

### PR DESCRIPTION
**Pull request by CatGarab**
The API, HAL, and lib readmes tend to be a bit light. I couldn't decide if there was anything super useful to put in them.

For the services, I mostly copied what's in each service's `main.rs` file

### CODE REVIEW

File: services/shell-service/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Shell Service
```
Comment (jacoffey3) [inline services/shell-service/README.md]: This should be entirely redundant and should be put in the official docs and linked to.

File: services/telemetry-service/README.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
+# Telemetry Database Service
```
Comment (jacoffey3) [inline services/telemetry-service/README.md]: This should be entirely redundant and should be put in the official docs and linked to.

**Comment by jacoffey3** [comment]
[READMEs in Kubos Repo](https://trello.com/c/82uNJDPZ/1017-readmes-in-kubos-repo)

File: clients/uart-comms-client/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Kubos UART Comms Service Client
```
[the remark this answers is not in the document]
Comment (CatGarab) [reply clients/uart-comms-client/README.md]: 1. It corresponds with our example `serial-comms-service` and is used in the comms service tutorial. It gives an example comms service over an actual hardware link
2. Yes, it was updated to be compatible with the spacepacket changes

**Pull request merged**